avm/res/app/container-app/tests/e2e/max/dependencies.bicep

```
@@ -17,16 +27,60 @@ resource managedEnvironment 'Microsoft.App/managedEnvironments@2023-05-01' = {
   }
 }
 
+resource keyVault 'Microsoft.KeyVault/vaults@2023-07-01' = {
+  name: keyVaultName
+  location: location
+  properties: {
+    sku: {
+      family: 'A'
+      name: 'standard'
+    }
+    tenantId: subscription().tenantId
+    publicNetworkAccess: 'Enabled'
+    enableRbacAuthorization: true
+  }
+}
+
+resource keyVaultSecret 'Microsoft.KeyVault/vaults/secrets@2023-07-01' = {
+  parent: keyVault
+  name: keyVaultSecretName
+  properties: {
+    value: myCustomKeyVaultSecret
+  }
+}
```

```suggestion
resource keyVault 'Microsoft.KeyVault/vaults@2023-07-01' = {
  name: keyVaultName
  location: location
  properties: {
    sku: {
      family: 'A'
      name: 'standard'
    }
    tenantId: subscription().tenantId
    publicNetworkAccess: 'Enabled'
    enableRbacAuthorization: true
  }

  resource secret 'secrets@2023-07-01' = {
    parent: keyVault
    name: 'my-secret'
    properties: {
      value: myCustomKeyVaultSecret
    }
  }
}
```

This enables you to remove the `keyVaultSecretName` parameter, and just requires you to update the output to

```bicep
@description('The key vault secret URI.')
output keyVaultSecretURI string = keyVault::secret.properties.secretUri
```

---

Hey @AlexanderSehr, thanks for the review! If it's okay, I think I'll just keep it as is for consistency.

---

I did remove the main.json change to avoid the version change. Thanks for the heads up with that!